File: src/main/java/com/jaffa/rpc/lib/exception/JaffaRpcNoRouteException.java
package com.jaffa.rpc.lib.exception;

import com.jaffa.rpc.lib.entities.Protocol;

import java.util.Objects;

public class JaffaRpcNoRouteException extends RuntimeException {
    private static final String MESSAGE_PREFIX = "No route for service: ";

    public JaffaRpcNoRouteException(String service, String moduleId) {
        super(MESSAGE_PREFIX + service + (Objects.nonNull(moduleId) ? (" and module.id " + moduleId) : ""));
    }

    public JaffaRpcNoRouteException(String service) {
        super(MESSAGE_PREFIX + service);
    }

    public JaffaRpcNoRouteException(String service, Protocol protocol) {
        super(MESSAGE_PREFIX + service + " and protocol " + protocol.getShortName());
    }
}

File: src/main/kotlin/com/jaffa/rpc/lib/exception/JaffaRpcNoRouteException.kt
package com.jaffa.rpc.lib.exception

import com.jaffa.rpc.lib.entities.Protocol

class JaffaRpcNoRouteException : RuntimeException {
    constructor(
        service: String?,
        moduleId: String?
    ) : super(MESSAGE_PREFIX + service + if (moduleId != null) " and module.id $moduleId" else "")

    constructor(service: String?) : super(MESSAGE_PREFIX + service)
    constructor(
        service: String?,
        protocol: Protocol
    ) : super(MESSAGE_PREFIX + service + " and protocol " + protocol.shortName)

    companion object {
        private const val MESSAGE_PREFIX = "No route for service: "
    }
}